src/main/java/org/example/bookstore/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,153 @@
+    public OrderResponseDto createNewOrder(OrderRequestDto requestDto,
+                                           Long userId) {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException("User with id: " + userId
+                        + "does not exist"));
+        ShoppingCart shoppingCart = shoppingCartRepository
+                .findShoppingCartByUserId(user.getId()).orElseThrow(()
+                        -> new EntityNotFoundException("Shopping cart does not exist"));
+        Set<CartItem> cartItems = shoppingCart.getCartItems();
+
+        if (cartItems == null || cartItems.isEmpty()) {
+            throw new EmptyCartException("Your shopping cart is empty ");
+        }
+
+        Order order = new Order();
+        order.setUser(user);
+        order.setOrderDate(LocalDateTime.now());
+        order.setStatus(Status.PENDING);
+        order.setShippingAddress(requestDto.getShippingAddress());
+
+        Set<OrderItem> orderItems = cartItems.stream().map(cartItem -> {
+            OrderItem orderItem = new OrderItem();
+            orderItem.setOrder(order);
+
+            orderItem.setBook(cartItem.getBook());
+            orderItem.setQuantity(cartItem.getQuantity());
+            orderItem.setPrice(cartItem.getBook().getPrice().multiply(
+                    BigDecimal.valueOf(cartItem.getQuantity())));
+
+            return orderItem;
+        }).collect(Collectors.toSet());
+        order.setOrderItems(orderItems);
+        order.setTotal(orderItems.stream()
+                .map(orderItem -> orderItem.getPrice().multiply(
+                        BigDecimal.valueOf(orderItem.getQuantity())))
+                .reduce(BigDecimal.ZERO, BigDecimal::add));
+
+        Order savedOrder = orderRepository.save(order);
+        orderItemRepository.saveAll(orderItems);
+        cartItemRepository.deleteAllByShoppingCartId(shoppingCart.getId());
+        return orderMapper.toDto(savedOrder);
+    }
+
+    @Override
+    @Transactional
+    public List<OrderResponseDto> getAllUserOrders(Long userId,Pageable pageable) {
+        User user = userRepository.findById(userId).orElseThrow(() -> new EntityNotFoundException(
+                "can`t find user"));
+        List<Order> ordersByUserId = orderRepository.findAllByUserId(user.getId());
+        return ordersByUserId.stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    @Transactional
+    public List<OrderItemResponseDto> getAllItemsByOrderId(Long orderId) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() -> new EntityNotFoundException("can`t find "
+                        + "order by id: " + orderId));
+        Set<OrderItem> orderItems = order.getOrderItems();
+
+        return orderItems.stream()
+                .map(orderItemsMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    @Transactional
```

Comment:
following up with @yevheniisynytsia comment, we can remove `@transactional` from the get operations, or set `readOnly = true` for them